ci/build.sh

```diff
@@ -151,61 +151,40 @@ if [ -n "$GITHUB_TOKEN" ] && [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
+function push_docker_image() {
+  local version="$1"
+
+  echo "Tagging docker image $DOCKER_TAG:local with $DOCKER_TAG:$version..."
+  docker tag $DOCKER_TAG:local $DOCKER_TAG:$version
+  echo "Pushing docker image $DOCKER_TAG:$version..."
   echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-  docker push $DOCKER_TAG:$VERSION
+  docker push $DOCKER_TAG:$version
+}
+
+function push_helm_chart() {
+  local version="$1"
 
-  # Push new helm release
-  echo "Pushing new Helm Chart release $VERSION"
+  echo "Pushing new Helm Chart release $version"
   git checkout -- .
   sudo helm init --client-only
-  sudo helm package deploy/helm/sumologic --dependency-update --version=$VERSION --app-version=$VERSION
+  sudo helm package deploy/helm/sumologic --dependency-update --version=$version --app-version=$version
   git fetch origin-repo
   git checkout gh-pages
   sudo helm repo index ./ --url https://sumologic.github.io/sumologic-kubernetes-collection/
   git add -A
-  git commit -m "Push new Helm Chart release $VERSION"
+  git commit -m "Push new Helm Chart release $version"
   git push --quiet origin-repo gh-pages
+}
 
-elif [ -n "$DOCKER_PASSWORD" ] && [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_EVENT_TYPE" == "push" ]; then
-  # Major.minor.patch version format
-  latest_release=$(wget -q $DOCKER_TAGS -O - | jq -r .[].name | grep -v alpha | grep -v latest | sort --version-sort --field-separator=. | tail -1)
-  latest_major=$(echo $latest_release | tr '.' $'\n' | sed -n 1p)
-  latest_minor=$(echo $latest_release | tr '.' $'\n' | sed -n 2p)
-  latest_alpha=$(wget -q $DOCKER_TAGS -O - | jq -r .[].name | grep alpha | grep "^$latest_major.$latest_minor" | sed 's/-alpha//g' | sort --version-sort --field-separator=. | tail -1)
-  if [ -n "$latest_alpha" ]; then
-    echo "Most recent release version: $latest_release, most recent alpha: $latest_alpha-alpha"
-  else
-    echo "Most recent release version: $latest_release, most recent alpha does not yet exist"
-    latest_alpha="$latest_release"
-  fi
-  new_patch=$(($(echo $latest_alpha | tr '.' $'\n' | sed -n 3p)+1))
-  new_alpha="$latest_major.$latest_minor.$new_patch-alpha"
-  
-  echo "Tagging docker image $DOCKER_TAG:local with $DOCKER_TAG:$new_alpha..."
-  docker tag $DOCKER_TAG:local $DOCKER_TAG:$new_alpha
-  echo "Pushing alpha docker image with version $new_alpha"
-  echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-  docker push $DOCKER_TAG:$new_alpha
+if [ -n "$DOCKER_PASSWORD" ] && [ -n "$TRAVIS_TAG" ]; then
+  push_docker_image "$VERSION"
+  push_helm_chart "$VERSION"
 
-  echo "Tagging git with v$new_alpha..."
-  git tag -a "v$new_alpha" -m "Bump version to v$new_alpha"
-  git push --tags --quiet --set-upstream origin-repo master
+elif [ -n "$DOCKER_PASSWORD" ] && [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_EVENT_TYPE" == "push" ]; then
+  dev_build_tag=$(git describe --tags --always)
```

Comment:
if I'm merging a commit from a branch that predates one or more tags that have been created since it was branched, will this return the correct tag?

Reply:
Yeah, it works, as per doc: `The command finds the most recent tag that is reachable from a commit.`
https://git-scm.com/docs/git-describe

I've just checked it locally and it's working as described for merged branches as well.